openslides_backend/models/check_models.py

```diff
@@ -0,0 +1,690 @@
+        if self.is_template_field(foreign_field):
+            if replacement:
+                actual_foreign_field = cast(
+                    BaseTemplateField, foreign_field_type
+                ).get_structured_field_name(replacement)
+            else:
+                replacement_collection = cast(
+                    BaseTemplateField, foreign_field_type
+                ).replacement_collection
+                if replacement_collection:
+                    replacement = model.get(f"{replacement_collection.collection}_id")
+                if not replacement:
+                    self.errors.append(
+                        f"{basemsg} points to {foreign_collection}/{foreign_id}/{foreign_field},"
+                        f" but there is no replacement for {replacement_collection}"
+                    )
+                actual_foreign_field = self.make_structured(
+                    cast(BaseTemplateField, foreign_field_type), replacement
+                )
+
+        foreign_model = self.find_model(foreign_collection, foreign_id)
+        foreign_value = (
+            foreign_model.get(actual_foreign_field)
+            if foreign_model is not None
+            else None
+        )
+        fqid = f"{collection}/{id}"
+        error = False
+        if isinstance(foreign_field_type, RelationField):
+            error = foreign_value != id
+        elif isinstance(foreign_field_type, RelationListField):
+            error = not foreign_value or id not in foreign_value
+        elif isinstance(foreign_field_type, GenericRelationField):
+            error = foreign_value != fqid
+        elif isinstance(foreign_field_type, GenericRelationListField):
+            error = not foreign_value or fqid not in foreign_value
+        else:
+            raise NotImplementedError()
+
+        if error:
+            self.errors.append(
+                f"{basemsg} points to {foreign_collection}/{foreign_id}/{actual_foreign_field},"
+                " but the reverse relation for is corrupt"
+            )
+
+    def split_fqid(self, fqid: str) -> Tuple[str, int]:
+        try:
+            collection, _id = fqid.split("/")
+            id = int(_id)
+            if collection not in self.models.keys():
+                raise CheckException(f"Fqid {fqid} has an invalid collection")
+            return collection, id
+        except (ValueError, AttributeError):
+            raise CheckException(f"Fqid {fqid} is malformed")
+
+    def split_collectionfield(self, collectionfield: str) -> Tuple[str, str]:
+        collection, field = collectionfield.split("/")
+        if collection not in self.models.keys():
+            raise CheckException(
+                f"Collectionfield {collectionfield} has an invalid collection"
+            )
+        if field not in [
+            field.own_field_name for field in self.models[collection]().get_fields()
+        ]:  # Note: this has to be adopted when supporting template fields
```

Comment:
Please resolve this TODO: for template fields, `get_template_field_name` must be used (maybe you should add a util method like `get_own_field_name` or similar, since this pattern seems to occur more often)

Reply:
Add a function `get_own_field_name` and use it instead `field.own_field_name`

Reply:
Please remove the TODO then

Reply:
Removed